### PR TITLE
Added NXOS provider for tacacs

### PR DESCRIPTION
**Pull request by jonnytpuppet**

### CODE REVIEW

**Comment by jonnytpuppet** [comment]
tests/beaker_tests/tacacs/tacacs_provider_defaults.rb passed in 33.99 seconds
      Test Suite: tests @ 2015-11-12 14:02:49 +0000

**Comment by robert-w-gries** [comment]
Please add this line to examples/demo_all.pp

`include ciscopuppet::demo_tacacs`

File: README.md [hunk 1]
```diff
@@ -196,7 +196,11 @@ The following resources include cisco types and providers along with cisco provi
   * [`cisco_tacacs_server`](#type-cisco_tacacs_server)
   * [`tacacs_server_group (netdev_stdlib)`](#type-cisco_tacacs_server_group)
   * [`cisco_tacacs_server_host`](#type-cisco_tacacs_server_host)
+<<<<<<< HEAD
```
Comment (robert-w-gries) [inline README.md]: Leftover merge conflicts from your develop => add_tacas merge

Comment (jonnytpuppet) [inline README.md]: @robert-w-gries we're not adding these nedev demos to demo_all. Please speak to @mikewiebe on future plans to come in another PR.

I'll fix the merge conflicts now

**Comment by jonnytpuppet** [comment]
@robert-w-gries @hunsbea merge artifacts removed

**Comment by jonnytpuppet** [comment]
@mikewiebe @hunsbea @robert-w-gries Can this be merged now? Please ignore if the README is a little off, I'll fix it in another merge. It's getting difficult to fix the conflicts 100%.

**Comment by robert-w-gries** [comment]
:+1:

**Pull request merged**